File: backend/Models/mastering/Mastering.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics;

namespace backend.Models {

  public enum Level {
    Beginner, Intermediate, Advanced, Expert
  }

  public class Mastering : IValidatableObject {
    
    [Key]
    public int masteringId { get; set; }
   [ForeignKey(nameof(User))]
    public int userId { get; set; }
    public User User { get; set; }

    [ForeignKey(nameof(Skill))]
    public int SkillId { get; set; }
    public Skill Skill { get; set; }

  
    public Level Level { get; set; }

     public IEnumerable<ValidationResult> Validate(ValidationContext validationContext) {
            var currContext = validationContext.GetService(typeof(MainContext)) as MainContext;
            Debug.Assert(currContext != null);
            yield return new ValidationResult(null);
            
    }





      

  }

}

File: frontend/src/app/models/Mastering.ts


import { ParseErrorLevel } from "@angular/compiler";
import * as internal from "assert";
import { Transform, Type } from "class-transformer";
import * as moment from "moment";
import { Moment } from "moment";

import 'reflect-metadata';
import { Category } from "./Category";
import { Skill } from "./Skill";


export enum Level {
    Beginner = 0, Intermediate = 1, Advanced = 2, Expert=3
}


export class Mastering {

   masteringId?: number;
   userId?: number;
   skillId?: number;
   skill?: Skill;
   level?: Level;

   public get skilltoString(){
       return this.skill?.name;
   }
   public get categoryToString(){
       return this.skill?.category?.name;
   }

   public set changeLevel(level : number) {
       this.level = level;
   }

   public get levelValue(){
       if(this.level?.toString() === "Beginner"){
           return 0;
       }
       else if(this.level?.toString() == "Intermediate"){
           return 1;
       }
       else {
           return 2;
       }
   }

 

}